src/prefect/tasks/databricks/databricks_hook.py

```diff
@@ -333,6 +335,37 @@ def terminate_cluster(self, json: dict) -> None:
         """
         self._do_api_call(TERMINATE_CLUSTER_ENDPOINT, json)
 
+    def get_job_id_by_name(self, job_name: str, limit: int = 25) -> int:
+        """
+        Retrieves job_id from run_id.
+
+        Args:
+            - job_name (str): Name of the job.
+
+        Returns
+            - int: Job id for given Databricks job.
+        """
+        matching_jobs = []
+        while True:
+            response_payload = self._do_api_call(LIST_JOB_ENDPOINT, {"limit": limit})
```

Comment:
I think you also need to include an offset in this API call, otherwise you'll fetch the same jobs repeatedly in an infinite loop.

Reply:
Thanks for picking this up! Definitely fell asleep on this one, made this change and the one requested below.